File: mine-mart-fe/src/types/supplier.ts
/**
 * Supplier types - maps to backend DTOs
 */

export interface SupplierResponse {
  id: number;
  name: string;
  phone?: string;
  email?: string;
  address?: string;
  note?: string;
}

export interface SupplierCreateRequest {
  name: string;
  phone?: string;
  email?: string;
  address?: string;
  note?: string;
}

export interface SupplierUpdateRequest {
  name?: string;
  phone?: string;
  email?: string;
  address?: string;
  note?: string;
}

File: mini-mart/src/main/java/com/example/mini_mart/entity/Supplier.java
package com.example.mini_mart.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "supplier")
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 500)
    @NotNull
    @Column(name = "name", nullable = false, length = 500)
    private String name;

    @Size(max = 50)
    @Column(name = "phone", length = 50)
    private String phone;

    @Size(max = 255)
    @Column(name = "email")
    private String email;

    @Size(max = 500)
    @Column(name = "address", length = 500)
    private String address;

    @Size(max = 1000)
    @Column(name = "note", length = 1000)
    private String note;

    @OneToMany(mappedBy = "supplier")
    private List<ImportReceipt> importReceipts = new ArrayList<>();

}